File: sol/1261/1261.cpp
struct FindElements {
    TreeNode *m_root;
    FindElements(TreeNode* root):
        m_root(root) {}
    
    bool find(int target) {
        ++target;
        unsigned mask = 1;
        while (mask <= target){
            mask <<= 1;
        }
        mask >>= 2;
        TreeNode *ptr = m_root;
        while (ptr != nullptr && mask != 0){
            if (target & mask){
                ptr = ptr->right;
            } else {
                ptr = ptr->left;
            }
            mask >>= 1;
        }
        return ptr != nullptr;
    }
};

File: sol/1261/1261.py
class FindElements:
    def __init__(self, root: Optional[TreeNode]):
        self.root = root

    def find(self, target: int) -> bool:
        target += 1
        mask = 1
        while mask <= target:
            mask <<= 1
        mask >>= 2
        ptr = self.root
        while ptr is not None and mask != 0:
            if target & mask:
                ptr = ptr.right
            else:
                ptr = ptr.left
            mask >>= 1
        return ptr is not None

